msal/tests/Test.MSAL.NET.UIAutomation/MSALMobileTestHelper.cs

```diff
@@ -309,5 +325,21 @@ public void PerformB2CSignInFlow(ITestController controller, LabUser user, B2CId
             }
             CoreMobileTestHelper.VerifyResult(controller);
         }
+
+        public void PerformB2CSignInEditProfileFlow(ITestController controller, B2CIdentityProvider b2CIdentityProvider)
+        {
+            SetB2CInputDataForEditProfileAuthority(controller);
+            
+            controller.Tap(CoreUiTestConstants.AcquirePageID);
+            
+            SetUiBehavior(controller, CoreUiTestConstants.UIBehaviorNone);
```

Comment:
>SetUiBehavior(controller, CoreUiTestConstants.UIBehaviorNone); [](start = 12, length = 62)

Makes sense to ask JM if this is how he envisions the setup. Also... what if the user is not logged in? Currently thinking the service side has to be part of deciding whether UI is shown or not. Would be great to get JM's perspective, and our friends in B2C (e.g. P and J).

Reply:
if it's used as a silent call, server side will return this:
 `One of two conditions was encountered: 1. The UiBehavior.Never flag was passed, but the constraint could not be honored, because user interaction was required. 2. An error occurred during a silent web authentication that prevented the http authentication flow from completing in a short enough time frame` 

---
In reply to: [236440625](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1424#discussion_r236440625) [](ancestors = 236440625)

Reply:
Trying to clarify my train of thoughts. 

Yeah, this flow doesn't make sense for the silent flow. My question is if the user is not logged in and the first flow you call is the edit profile flow... I assume regardless of us passing in none, the user should still be met with a prompt dialog... this is why I'm asking for server ack of what we are doing. If they treat none as do not show the auth dialogs, then... it will require the developer to know if the user is logged in or not. 

---
In reply to: [236450550](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1424#discussion_r236450550) [](ancestors = 236450550,236440625)

Reply:
I see what you're saying. I thought you meant if the dev used prompt=none in another flow to mimic a silent call.
To answer your question, if the user is not signed in and does an edit profile w/prompt=none, we get this back from B2C: 
`- AADB2C90077: User does not have an existing session and request prompt parameter has a value of 'None'.`
if you do edit profile w/select account, you can select the account first (login) and then edit the profile.
so, yes, the dev will need to know if the user has logged in, or they can surface this after the user has logged in, I think that's how our samples are set up. You log-in first and then are given the option to edit profile.

---
In reply to: [236455405](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1424#discussion_r236455405) [](ancestors = 236455405,236450550,236440625)

Reply:
Great. As expected. It will though be hard as a developer to fully build the experience with-out keeping state separately... but perhaps that is good enough. We should have a wiki page explaining this (and the evo messages at time of writing keeping in mind that is all subject to change). Also would be good to investigate what the b2c service would prefer going forward that we send.

---
In reply to: [236458472](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1424#discussion_r236458472) [](ancestors = 236458472,236455405,236450550,236440625)

Reply:
sounds good. i'll follow up w/b2c

---
In reply to: [236462577](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1424#discussion_r236462577) [](ancestors = 236462577,236458472,236455405,236450550,236440625)